stylix/palette.nix

```diff
@@ -5,27 +5,88 @@
   options,
   ...
 }:
-
 let
   cfg = config.stylix;
-
 in
 {
+  imports = [
+    (lib.mkRemovedOptionModule
+      [
+        "stylix"
+        "polarity"
+      ]
+      "stylix: `stylix.polarity` has been removed in favor of `stylix.themeGeneration.polarity`, and its `either` value is no longer available"
+    )
+  ];
+
   options.stylix = {
-    polarity = lib.mkOption {
-      type = lib.types.enum [
-        "either"
-        "light"
-        "dark"
-      ];
-      default = "either";
-      description = ''
-        Use this option to force a light or dark theme.
+    themeGeneration = {
```

Comment:
It would be better to replace the `lib.types.float` type with a type that only allows `float` between `-1` and `1`. That way, this explanation can be removed from the description of the options.

Reply:
Adding the type constraint moves the runtime check to evaluation time. Either way, this will be included in the documentation. For example, here is what the Home Manager documentation looks like:

![image](https://github.com/user-attachments/assets/ade728b4-aa9c-4ea4-b58f-dfc1f99d8c9b)